### PR TITLE
Copy `cortex/pkg/compactor` requirements into Tempo

### PR DESCRIPTION
**Pull request by zalegrala**
**What this PR does**:

As part of the migration away from cortex, here we copy what we need
to satisfy the compactor.

### CODE REVIEW

File: modules/compactor/config.go [hunk 1]
```diff
@@ -5,7 +5,8 @@ import (
 	"fmt"
 	"time"
 
-	cortex_compactor "github.com/cortexproject/cortex/pkg/compactor"
+	tempo_compactor "github.com/grafana/tempo/pkg/compactor"
```
Comment (joe-elliott) [inline modules/compactor/config.go:8]: remove alias. `compactor` is fine.

**Review by joe-elliott — COMMENTED** [review]
one nit. looks good.

**Review by mapno — APPROVED** [review]

File: modules/compactor/config.go (hunk 1 above)
Comment (joe-elliott) [reply modules/compactor/config.go]: Oh, I see, you moved it to `/pkg/compactor``. Let' s just move this code directly into `/modules/compactor`. It's only used by that module and we'd prefer keeping it isolated there to making a new module.

Generally we only put things in `/pkg` that will be shared by multiple components of Tempo.

Comment (zalegrala) [reply modules/compactor/config.go]: Sure thing, I've updated to keep it all in the `modules/`.  For the distributor PR also.

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**